velox/exec/tests/DriverTest.cpp

```diff
@@ -818,6 +826,294 @@ TEST_F(DriverTest, driverCreationThrow) {
       "Can only create 1 'throw driver'.");
 }
 
+DEBUG_ONLY_TEST_F(DriverTest, driverSuspensionRaceWithTaskPause) {
+  std::vector<RowVectorPtr> batches;
+  for (int i = 0; i < 4; ++i) {
+    batches.push_back(
+        makeRowVector({"c0"}, {makeFlatVector<int32_t>({1, 2, 3})}));
+  }
+
+  struct {
+    int numDrivers;
+    bool enterSuspensionAfterPauseStarted;
+    StopReason expectedEnterSuspensionStopReason;
+    bool leaveSuspensionDuringPause;
+    StopReason expectedLeaveSuspensionStopReason;
+
+    std::string debugString() const {
+      return fmt::format(
+          "numDrivers:{} enterSuspensionAfterPauseStarted:{} expectedEnterSuspensionStopReason:{} leaveSuspensionDuringPause:{} expectedLeaveSuspensionStopReason:{}",
+          numDrivers,
+          enterSuspensionAfterPauseStarted,
+          expectedEnterSuspensionStopReason,
+          leaveSuspensionDuringPause,
+          expectedLeaveSuspensionStopReason);
+    }
+  } testSettings[] = {
+      {1, true, StopReason::kNone, true, StopReason::kNone},
```

Comment:
nit: It would be nice to add comments to explain what each test cases is for.

Reply:
Commented.